src/graphics/program-lib/programs/standard.js

```diff
@@ -155,6 +155,18 @@ const standard = {
 
             if (encoding) {
                 subCode = subCode.replace(/\$DECODE/g, ChunkUtils.decodeFunc((!options.gamma && encoding === 'srgb') ? 'linear' : encoding));
+
+                // continue to support $texture2DSAMPLE
+                if (subCode.indexOf('$texture2DSAMPLE')) {
+                    const decodeTable = {
```

Comment:
keep this object at the global scope?

Reply:
I am convinced the compiler does not construct/allocate this object every time the function is called (but rather hoists it globally or something similar - equivalent to c++ const block). Having it declared here where it is used is much clearer for the reader.

Now just need to find some confirmation of my conviction on the webs...